--- panda.orig/arch/arm/Makefile
+++ panda/arch/arm/Makefile
@@ -28,7 +28,7 @@
 ifneq ($(CROSS_COMPILE),)
 KBUILD_CPPFLAGS	+= -target $(CROSS_COMPILE:-=)
 endif
-KBUILD_CFLAGS   += -fno-builtin -Wno-asm-operand-widths -Xassembler -mno-warn-deprecated
+KBUILD_CFLAGS   += -fno-builtin -Wno-asm-operand-widths
 endif
 
 # Never generate .eh_frame
@@ -45,7 +45,7 @@
 endif
 
 ifeq ($(CONFIG_FRAME_POINTER),y)
-KBUILD_CFLAGS	+=-fno-omit-frame-pointer -mapcs -mno-sched-prolog
+KBUILD_CFLAGS	+=-fno-omit-frame-pointer
 endif
 
 ifeq ($(CONFIG_CPU_BIG_ENDIAN),y)
