Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Index: linux/arch/arm/Makefile
===================================================================
--- linux.orig/arch/arm/Makefile	2012-09-05 12:03:39.809231310 -0400
+++ linux/arch/arm/Makefile	2012-09-05 12:13:34.565754207 -0400
@@ -49,7 +49,7 @@
 AS		+= -EB
 LD		+= -EB
 else
-KBUILD_CPPFLAGS	+= -mlittle-endian
+#KBUILD_CPPFLAGS	+= -mlittle-endian
 AS		+= -EL
 LD		+= -EL
 endif
@@ -93,7 +93,7 @@
 tune-$(CONFIG_CPU_V6K)		:=$(call cc-option,-mtune=arm1136j-s,-mtune=strongarm)
 
 ifeq ($(CONFIG_AEABI),y)
-CFLAGS_ABI	:=-mabi=aapcs-linux -mno-thumb-interwork
+CFLAGS_ABI	:=
 else
 CFLAGS_ABI	:=$(call cc-option,-mapcs-32,-mabi=apcs-gnu) $(call cc-option,-mno-thumb-interwork,)
 endif
@@ -114,7 +114,7 @@
 endif
 
 # Need -Uarm for gcc < 3.x
-KBUILD_CFLAGS	+=$(CFLAGS_ABI) $(CFLAGS_THUMB2) $(arch-y) $(tune-y) $(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,)) -msoft-float -Uarm
+KBUILD_CFLAGS	+=$(CFLAGS_ABI) $(CFLAGS_THUMB2) $(arch-y) $(tune-y) -msoft-float
 KBUILD_AFLAGS	+=$(CFLAGS_ABI) $(AFLAGS_THUMB2) $(arch-y) $(tune-y) -include asm/unified.h -msoft-float
 
 CHECKFLAGS	+= -D__arm__
Index: linux/arch/arm/include/asm/ftrace.h
===================================================================
--- linux.orig/arch/arm/include/asm/ftrace.h	2012-09-05 12:03:39.813231260 -0400
+++ linux/arch/arm/include/asm/ftrace.h	2012-09-05 12:13:34.569754156 -0400
@@ -45,7 +45,7 @@
 
 #else
 
-extern inline void *return_address(unsigned int level)
+extern inline __attribute__((gnu_inline)) void *return_address(unsigned int level)
 {
 	return NULL;
 }
Index: linux/include/linux/netdevice.h
===================================================================
--- linux.orig/include/linux/netdevice.h	2012-09-05 12:03:39.837230959 -0400
+++ linux/include/linux/netdevice.h	2012-09-05 12:13:34.569754156 -0400
@@ -1266,7 +1266,7 @@
 	enum {
 		RTNL_LINK_INITIALIZED,
 		RTNL_LINK_INITIALIZING,
-	} rtnl_link_state:16;
+	} rtnl_link_state;
 
 	/* Called from unregister, can be used to call free_netdev */
 	void (*destructor)(struct net_device *dev);
