--- linux.orig/arch/arm/include/asm/thread_info.h
+++ linux/arch/arm/include/asm/thread_info.h
@@ -88,6 +88,16 @@
 #define init_thread_info	(init_thread_union.thread_info)
 #define init_stack		(init_thread_union.stack)
 
+
+/*
+ * how to get the current stack pointer from C
+ */
+#define current_stack_pointer ({ \
+	unsigned long current_sp; \
+	asm ("mov %0, r13" : "=r" (current_sp)); \
+	current_sp; \
+})
+
 /*
  * how to get the thread information struct from C
  */
