--- linux-gcc.orig/arch/arm/include/asm/percpu.h
+++ linux-gcc/arch/arm/include/asm/percpu.h
@@ -30,14 +30,15 @@
 static inline unsigned long __my_cpu_offset(void)
 {
 	unsigned long off;
-	register unsigned long *sp asm ("sp");
+	unsigned long sp;
+	asm("mov %0, r13" : "=r" (sp));
 
 	/*
 	 * Read TPIDRPRW.
 	 * We want to allow caching the value, so avoid using volatile and
 	 * instead use a fake stack read to hazard against barrier().
 	 */
-	asm("mrc p15, 0, %0, c13, c0, 4" : "=r" (off) : "Q" (*sp));
+	asm("mrc p15, 0, %0, c13, c0, 4" : "=r" (off) : "Q" (sp));
 
 	return off;
 }
