Index: linux/arch/arm/include/asm/thread_info.h
===================================================================
--- linux.orig/arch/arm/include/asm/thread_info.h	2012-09-05 12:03:39.813231260 -0400
+++ linux/arch/arm/include/asm/thread_info.h	2012-09-05 12:13:43.297644753 -0400
@@ -93,8 +93,7 @@
 
 static inline struct thread_info *current_thread_info(void)
 {
-	register unsigned long sp asm ("sp");
-	return (struct thread_info *)(sp & ~(THREAD_SIZE - 1));
+	return (struct thread_info *)((unsigned long)(__builtin_frame_address(0)) & ~(THREAD_SIZE - 1));
 }
 
 #define thread_saved_pc(tsk)	\
Index: linux/arch/arm/kernel/return_address.c
===================================================================
--- linux.orig/arch/arm/kernel/return_address.c	2012-09-05 12:03:39.813231260 -0400
+++ linux/arch/arm/kernel/return_address.c	2012-09-05 12:13:43.297644753 -0400
@@ -39,12 +39,11 @@
 {
 	struct return_address_data data;
 	struct stackframe frame;
-	register unsigned long current_sp asm ("sp");
 
 	data.level = level + 1;
 
 	frame.fp = (unsigned long)__builtin_frame_address(0);
-	frame.sp = current_sp;
+	asm ("mov %0, r13" : "=r" (frame.sp));
 	frame.lr = (unsigned long)__builtin_return_address(0);
 	frame.pc = (unsigned long)return_address;
 
Index: linux/arch/arm/kernel/stacktrace.c
===================================================================
--- linux.orig/arch/arm/kernel/stacktrace.c	2012-09-05 12:03:39.813231260 -0400
+++ linux/arch/arm/kernel/stacktrace.c	2012-09-05 12:13:43.297644753 -0400
@@ -109,11 +109,9 @@
 		frame.pc = thread_saved_pc(tsk);
 #endif
 	} else {
-		register unsigned long current_sp asm ("sp");
-
 		data.no_sched_functions = 0;
 		frame.fp = (unsigned long)__builtin_frame_address(0);
-		frame.sp = current_sp;
+		asm ("mov %0, r13" : "=r" (frame.sp));
 		frame.lr = (unsigned long)__builtin_return_address(0);
 		frame.pc = (unsigned long)save_stack_trace_tsk;
 	}
Index: linux/arch/arm/kernel/unwind.c
===================================================================
--- linux.orig/arch/arm/kernel/unwind.c	2012-09-05 12:03:39.813231260 -0400
+++ linux/arch/arm/kernel/unwind.c	2012-09-05 12:13:43.297644753 -0400
@@ -408,7 +408,6 @@
 void unwind_backtrace(struct pt_regs *regs, struct task_struct *tsk)
 {
 	struct stackframe frame;
-	register unsigned long current_sp asm ("sp");
 
 	pr_debug("%s(regs = %p tsk = %p)\n", __func__, regs, tsk);
 
@@ -424,7 +423,7 @@
 			 ? regs->ARM_pc : regs->ARM_lr;
 	} else if (tsk == current) {
 		frame.fp = (unsigned long)__builtin_frame_address(0);
-		frame.sp = current_sp;
+		asm ("mov %0, r13" : "=r" (frame.sp));
 		frame.lr = (unsigned long)__builtin_return_address(0);
 		frame.pc = (unsigned long)unwind_backtrace;
 	} else {
