From a32a61a7bcfc8263beabf9bd080341854db7e2d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Mon, 2 Jul 2012 13:56:24 +0200
Subject: [PATCH 15/48] Disable clang's integrated-as for some vdso targets  . Patch from series at http://lists.cs.uiuc.edu/pipermail/llvm-commits/Week-of-Mon-20120507/142707.html by PaX Team.

---
 arch/x86/vdso/Makefile |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/x86/vdso/Makefile b/arch/x86/vdso/Makefile
index 5d17950..5b03c3c 100644
--- a/arch/x86/vdso/Makefile
+++ b/arch/x86/vdso/Makefile
@@ -29,6 +29,12 @@ VDSO_LDFLAGS_vdso.lds = -m64 -Wl,-soname=linux-vdso.so.1 \
 			-Wl,--no-undefined \
 		      	-Wl,-z,max-page-size=4096 -Wl,-z,common-page-size=4096
 
+AFLAGS_vdso.o = -no-integrated-as
+AFLAGS_vdso32.o = -no-integrated-as
+AFLAGS_int80.o = -no-integrated-as
+AFLAGS_syscall.o = -no-integrated-as
+AFLAGS_sysenter.o = -no-integrated-as
+
 $(obj)/vdso.o: $(src)/vdso.S $(obj)/vdso.so
 
 $(obj)/vdso.so.dbg: $(src)/vdso.lds $(vobjs) FORCE
-- 
1.7.3.4

