From 5fbcb904de5f4156cf7780f0405e7cf00758fc9b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Mon, 2 Jul 2012 14:22:04 +0200
Subject: [PATCH 16/48] Disable clang's integrated-as for arch/x86/core/kernel  . Patch from series at http://lists.cs.uiuc.edu/pipermail/llvm-commits/Week-of-Mon-20120507/142707.html by PaX Team.

---
 arch/x86/kernel/Makefile |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/x86/kernel/Makefile b/arch/x86/kernel/Makefile
index 5369059..bcc40ff 100644
--- a/arch/x86/kernel/Makefile
+++ b/arch/x86/kernel/Makefile
@@ -35,6 +35,10 @@ obj-y			+= tsc.o io_delay.o rtc.o
 obj-y			+= pci-iommu_table.o
 obj-y			+= resource.o
 
+AFLAGS_trampoline_32.o  = -no-integrated-as
+AFLAGS_trampoline_64.o  = -no-integrated-as
+AFLAGS_reboot_32.o  = -no-integrated-as
+
 obj-y				+= trampoline.o trampoline_$(BITS).o
 obj-y				+= process.o
 obj-y				+= i387.o xsave.o
-- 
1.7.3.4

