diff --git a/arch/arm/kernel/pmu.c b/arch/arm/kernel/pmu.c
index a4688e8..119488c 100644
--- a/arch/arm/kernel/pmu.c
+++ b/arch/arm/kernel/pmu.c
@@ -45,7 +45,7 @@ static int __devinit pmu_device_probe(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver cpu_pmu_driver = {
+static struct platform_driver cpu_pmu_driver __devinitdata= {
 	.driver		= {
 		.name	= "cpu-arm-pmu",
 	},
diff --git a/arch/arm/mach-msm/cpufreq.c b/arch/arm/mach-msm/cpufreq.c
index 2477221..8bcd2f8 100644
--- a/arch/arm/mach-msm/cpufreq.c
+++ b/arch/arm/mach-msm/cpufreq.c
@@ -285,7 +285,7 @@ static struct freq_attr *msm_freq_attr[] = {
 	NULL,
 };
 
-static struct cpufreq_driver msm_cpufreq_driver = {
+static struct cpufreq_driver msm_cpufreq_driver __initdata = {
 	/* lps calculations are handled here. */
 	.flags		= CPUFREQ_STICKY | CPUFREQ_CONST_LOOPS,
 	.init		= msm_cpufreq_init,
diff --git a/arch/arm/mach-msm/footswitch-8x60.c b/arch/arm/mach-msm/footswitch-8x60.c
index 98447fa..ff2710d 100644
--- a/arch/arm/mach-msm/footswitch-8x60.c
+++ b/arch/arm/mach-msm/footswitch-8x60.c
@@ -561,7 +561,7 @@ static int __devexit footswitch_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver footswitch_driver = {
+static struct platform_driver footswitch_driver __refdata = {
 	.probe		= footswitch_probe,
 	.remove		= __devexit_p(footswitch_remove),
 	.driver		= {
diff --git a/arch/arm/mach-msm/footswitch-pcom.c b/arch/arm/mach-msm/footswitch-pcom.c
index 8eec807..ed7f0c6 100644
--- a/arch/arm/mach-msm/footswitch-pcom.c
+++ b/arch/arm/mach-msm/footswitch-pcom.c
@@ -291,7 +291,7 @@ static int __devexit footswitch_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver footswitch_driver = {
+static struct platform_driver footswitch_driver __refdata = {
 	.probe		= footswitch_probe,
 	.remove		= __devexit_p(footswitch_remove),
 	.driver		= {
diff --git a/arch/arm/mach-msm/msm_dsps.c b/arch/arm/mach-msm/msm_dsps.c
index efb2bf6..057665b 100644
--- a/arch/arm/mach-msm/msm_dsps.c
+++ b/arch/arm/mach-msm/msm_dsps.c
@@ -868,7 +868,7 @@ static int __devexit dsps_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver dsps_driver = {
+static struct platform_driver dsps_driver __refdata = {
 	.probe          = dsps_probe,
 	.remove         = __exit_p(dsps_remove),
 	.driver         = {
diff --git a/arch/arm/mach-msm/pil-modem.c b/arch/arm/mach-msm/pil-modem.c
index 5aa3834..a8344c3 100644
--- a/arch/arm/mach-msm/pil-modem.c
+++ b/arch/arm/mach-msm/pil-modem.c
@@ -319,7 +319,7 @@ static int __devexit pil_modem_driver_exit(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver pil_modem_driver = {
+static struct platform_driver pil_modem_driver __refdata = {
 	.probe = pil_modem_driver_probe,
 	.remove = __devexit_p(pil_modem_driver_exit),
 	.driver = {
diff --git a/arch/arm/mach-msm/pil-q6v3.c b/arch/arm/mach-msm/pil-q6v3.c
index 7354d93..ffb9351 100644
--- a/arch/arm/mach-msm/pil-q6v3.c
+++ b/arch/arm/mach-msm/pil-q6v3.c
@@ -274,7 +274,7 @@ static int __devexit pil_q6v3_driver_exit(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver pil_q6v3_driver = {
+static struct platform_driver pil_q6v3_driver __refdata = {
 	.probe = pil_q6v3_driver_probe,
 	.remove = __devexit_p(pil_q6v3_driver_exit),
 	.driver = {
diff --git a/arch/arm/mach-msm/pil-tzapps.c b/arch/arm/mach-msm/pil-tzapps.c
index 90ac1d9..63f5d5a 100644
--- a/arch/arm/mach-msm/pil-tzapps.c
+++ b/arch/arm/mach-msm/pil-tzapps.c
@@ -71,7 +71,7 @@ static int __devexit pil_tzapps_driver_exit(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver pil_tzapps_driver = {
+static struct platform_driver pil_tzapps_driver __refdata = {
 	.probe = pil_tzapps_driver_probe,
 	.remove = __devexit_p(pil_tzapps_driver_exit),
 	.driver = {
diff --git a/arch/arm/mach-msm/rpm-regulator.c b/arch/arm/mach-msm/rpm-regulator.c
index 5c2bb8e..065ebfc 100644
--- a/arch/arm/mach-msm/rpm-regulator.c
+++ b/arch/arm/mach-msm/rpm-regulator.c
@@ -1393,7 +1393,7 @@ static int __devexit rpm_vreg_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver rpm_vreg_driver = {
+static struct platform_driver rpm_vreg_driver __refdata = {
 	.probe = rpm_vreg_probe,
 	.remove = __devexit_p(rpm_vreg_remove),
 	.driver = {
diff --git a/arch/arm/mach-msm/rpm.c b/arch/arm/mach-msm/rpm.c
index ef2956a..3fc36fb 100644
--- a/arch/arm/mach-msm/rpm.c
+++ b/arch/arm/mach-msm/rpm.c
@@ -819,7 +819,7 @@ static int __devexit msm_rpm_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver msm_rpm_platform_driver = {
+static struct platform_driver msm_rpm_platform_driver __refdata = {
 	.probe = msm_rpm_probe,
 	.remove = __devexit_p(msm_rpm_remove),
 	.driver = {
diff --git a/arch/arm/mach-msm/rpm_log.c b/arch/arm/mach-msm/rpm_log.c
index 3ed55da..df43a1a 100644
--- a/arch/arm/mach-msm/rpm_log.c
+++ b/arch/arm/mach-msm/rpm_log.c
@@ -335,7 +335,7 @@ static int __devexit msm_rpm_log_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver msm_rpm_log_driver = {
+static struct platform_driver msm_rpm_log_driver __refdata = {
 	.probe		= msm_rpm_log_probe,
 	.remove		= __devexit_p(msm_rpm_log_remove),
 	.driver		= {
diff --git a/arch/arm/mach-msm/rpm_stats.c b/arch/arm/mach-msm/rpm_stats.c
index a831bd5..f5d8888 100644
--- a/arch/arm/mach-msm/rpm_stats.c
+++ b/arch/arm/mach-msm/rpm_stats.c
@@ -222,7 +222,7 @@ static int __devexit msm_rpmstats_remove(struct platform_device *pdev)
 	platform_set_drvdata(pdev, NULL);
 	return 0;
 }
-static struct platform_driver msm_rpmstats_driver = {
+static struct platform_driver msm_rpmstats_driver __refdata = {
 	.probe	= msm_rpmstats_probe,
 	.remove = __devexit_p(msm_rpmstats_remove),
 	.driver = {
diff --git a/arch/arm/mach-msm/saw-regulator.c b/arch/arm/mach-msm/saw-regulator.c
index b11b1fa..edb7ca4 100644
--- a/arch/arm/mach-msm/saw-regulator.c
+++ b/arch/arm/mach-msm/saw-regulator.c
@@ -208,7 +208,7 @@ static int __devexit saw_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static struct platform_driver saw_driver = {
+static struct platform_driver saw_driver __refdata = {
 	.probe = saw_probe,
 	.remove = __devexit_p(saw_remove),
 	.driver = {
diff --git a/arch/arm/mach-msm/smd.c b/arch/arm/mach-msm/smd.c
index e794801..d6c60bd 100644
--- a/arch/arm/mach-msm/smd.c
+++ b/arch/arm/mach-msm/smd.c
@@ -2580,7 +2580,7 @@ static __init int modem_restart_late_init(void)
 }
 late_initcall(modem_restart_late_init);
 
-static struct platform_driver msm_smd_driver = {
+static struct platform_driver msm_smd_driver __initdata = {
 	.probe = msm_smd_probe,
 	.driver = {
 		.name = MODULE_NAME,
