diff --git a/arch/arm/mach-exynos/smc.S b/arch/arm/mach-exynos/smc.S
index 0449e57..db36efe 100644
--- a/arch/arm/mach-exynos/smc.S
+++ b/arch/arm/mach-exynos/smc.S
@@ -11,6 +11,7 @@
 #include <mach/smc.h>
 
 ENTRY(exynos_smc)
+	.arch_extension sec
 	stmfd	sp!, {lr}
 #if defined(CONFIG_ARM_ERRATA_766421)
 	dmb
diff --git a/arch/arm/mm/proc-v7.S b/arch/arm/mm/proc-v7.S
index 499eb34..ebfdc410 100644
--- a/arch/arm/mm/proc-v7.S
+++ b/arch/arm/mm/proc-v7.S
@@ -150,6 +150,7 @@ ENDPROC(cpu_v7_do_resume)
 #endif
 
 ENTRY(cpu_v7_setup_cp15)
+	.arch_extension sec
 	mrc	p15, 0, r0, c0, c0, 0		@ read main ID register
 	and	r10, r0, #0xff000000		@ ARM?
 	teq	r10, #0x41000000
