diff --git a/drivers/gpu/pvr/services4/srvkm/common/hash.c b/drivers/gpu/pvr/services4/srvkm/common/hash.c
index badcdb9..5417303 100644
--- a/drivers/gpu/pvr/services4/srvkm/common/hash.c
+++ b/drivers/gpu/pvr/services4/srvkm/common/hash.c
@@ -306,7 +306,6 @@ _Resize (HASH_TABLE *pHash, IMG_UINT32 uNewSize)
 #else
 	/* unused param */
 	*pHash = *pHash;
-	uNewSize = uNewSize;
 #endif
     return IMG_TRUE;
 }
