Index: msm/drivers/media/video/msm/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/Makefile	2012-09-18 12:34:08.458519025 -0700
+++ msm/drivers/media/video/msm/Makefile	2012-09-18 12:42:12.770527208 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ccflags-y += -Idrivers/media/video/msm/io
 ccflags-y += -Idrivers/media/video/msm/vfe
Index: msm/drivers/media/video/msm/actuators/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/actuators/Makefile	2012-09-17 13:18:18.909103837 -0700
+++ msm/drivers/media/video/msm/actuators/Makefile	2012-09-18 12:42:54.034527905 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/video/msm
 EXTRA_CFLAGS += -Idrivers/media/video/msm/io
 obj-$(CONFIG_MSM_ACTUATOR) += msm_actuator.o
Index: msm/drivers/media/video/msm/cci/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/cci/Makefile	2012-09-17 13:18:18.909103837 -0700
+++ msm/drivers/media/video/msm/cci/Makefile	2012-09-18 12:42:49.490527828 -0700
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/video/msm -Idrivers/media/video/msm/server
 obj-$(CONFIG_MSM_CCI) += msm_cci.o
Index: msm/drivers/media/video/msm/cpp/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/cpp/Makefile	2012-09-17 13:18:18.909103837 -0700
+++ msm/drivers/media/video/msm/cpp/Makefile	2012-09-18 12:41:58.158526961 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/video/msm
 ccflags-y += -Idrivers/media/video/msm/io
 obj-$(CONFIG_MSM_CPP) += msm_cpp.o
Index: msm/drivers/media/video/msm/csi/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/csi/Makefile	2012-09-17 13:18:18.909103837 -0700
+++ msm/drivers/media/video/msm/csi/Makefile	2012-09-18 12:41:50.074526826 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/video/msm -Idrivers/media/video/msm/server
 ifeq ($(CONFIG_MSM_CSI20_HEADER),y)
   ccflags-y += -Idrivers/media/video/msm/csi/include/csi2.0
Index: msm/drivers/media/video/msm/eeprom/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/eeprom/Makefile	2012-09-17 13:18:18.909103837 -0700
+++ msm/drivers/media/video/msm/eeprom/Makefile	2012-09-18 12:42:28.454527474 -0700
@@ -1,5 +1,5 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/video/msm/io
 obj-$(CONFIG_MSM_EEPROM) += msm_camera_eeprom.o
 obj-$(CONFIG_IMX074_EEPROM) += imx074_eeprom.o
-obj-$(CONFIG_IMX091_EEPROM) += imx091_eeprom.o
\ No newline at end of file
+obj-$(CONFIG_IMX091_EEPROM) += imx091_eeprom.o
Index: msm/drivers/media/video/msm/gemini/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/gemini/Makefile	2012-08-28 14:34:21.056125467 -0700
+++ msm/drivers/media/video/msm/gemini/Makefile	2012-09-18 12:42:06.070527098 -0700
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/video/msm
 obj-$(CONFIG_MSM_GEMINI) += msm_gemini_dev.o msm_gemini_sync.o msm_gemini_core.o msm_gemini_hw.o msm_gemini_platform.o
Index: msm/drivers/media/video/msm/io/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/io/Makefile	2012-09-17 13:18:18.913103838 -0700
+++ msm/drivers/media/video/msm/io/Makefile	2012-09-18 12:42:35.506527593 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ccflags-y += -Idrivers/media/video/msm -Idrivers/media/video/msm/cci
 obj-$(CONFIG_MSM_CAMERA)   += msm_camera_io_util.o msm_camera_i2c.o
Index: msm/drivers/media/video/msm/jpeg_10/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/jpeg_10/Makefile	2012-09-17 13:18:18.913103838 -0700
+++ msm/drivers/media/video/msm/jpeg_10/Makefile	2012-09-18 12:41:45.258526744 -0700
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/video/msm
 obj-$(CONFIG_MSM_JPEG) += msm_jpeg_dev.o msm_jpeg_sync.o msm_jpeg_core.o msm_jpeg_hw.o msm_jpeg_platform.o
Index: msm/drivers/media/video/msm/mercury/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/mercury/Makefile	2012-09-18 12:37:38.374522571 -0700
+++ msm/drivers/media/video/msm/mercury/Makefile	2012-09-18 12:39:43.278524686 -0700
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/video/msm
 obj-$(CONFIG_MSM_MERCURY) += msm_mercury_dev.o msm_mercury_core.o msm_mercury_hw.o msm_mercury_platform.o msm_mercury_sync.o
Index: msm/drivers/media/video/msm/sensors/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/sensors/Makefile	2012-09-17 13:18:18.917103838 -0700
+++ msm/drivers/media/video/msm/sensors/Makefile	2012-09-18 12:42:42.934527718 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/video/msm
 EXTRA_CFLAGS += -Idrivers/media/video/msm/io
 EXTRA_CFLAGS += -Idrivers/media/video/msm/eeprom
Index: msm/drivers/media/video/msm/server/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/server/Makefile	2012-09-17 13:18:18.921103837 -0700
+++ msm/drivers/media/video/msm/server/Makefile	2012-09-18 12:41:54.302526893 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ifeq ($(CONFIG_MSM_CAMERA_V4L2),y)
   EXTRA_CFLAGS += -Idrivers/media/video/msm
Index: msm/drivers/media/video/msm/vfe/Makefile
===================================================================
--- msm.orig/drivers/media/video/msm/vfe/Makefile	2012-09-17 13:18:18.921103837 -0700
+++ msm/drivers/media/video/msm/vfe/Makefile	2012-09-18 12:42:21.038527348 -0700
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/video/msm
 ccflags-y += -Idrivers/media/video/msm/server
 ifeq ($(GCC_VERSION),0404)
Index: msm/drivers/staging/rts5139/Makefile
===================================================================
--- msm.orig/drivers/staging/rts5139/Makefile	2012-09-17 13:18:19.541103848 -0700
+++ msm/drivers/staging/rts5139/Makefile	2012-09-18 12:43:18.682528322 -0700
@@ -27,7 +27,7 @@
 
 TARGET_MODULE := rts5139
 
-EXTRA_CFLAGS := -Idrivers/scsi -I$(PWD)
+EXTRA_CFLAGS := -Idrivers/scsi -I$(srctree)
 
 obj-m += $(TARGET_MODULE).o
 
