--- msm.orig/arch/arm/mach-msm/vreg.c
+++ msm/arch/arm/mach-msm/vreg.c
@@ -299,7 +299,7 @@
 }
 #else
 static inline int __init vreg_debug_init(void) { return 0; }
-static inline void __exit vreg_debug_exit(void) { return 0; }
+static inline void __exit vreg_debug_exit(void) { return; }
 #endif
 
 static int __init vreg_init(void)
--- msm.orig/arch/arm/mach-msm/perf_event_msm_krait_l2.c
+++ msm/arch/arm/mach-msm/perf_event_msm_krait_l2.c
@@ -18,6 +18,10 @@
 
 #include <mach/msm-krait-l2-accessors.h>
 
+#ifndef CONFIG_HW_PERF_EVENTS
+#error "CONFIG_HW_PERF_EVENTS is undefined but is required"
+#endif
+
 #define MAX_L2_PERIOD	((1ULL << 32) - 1)
 #define MAX_KRAIT_L2_CTRS 10
 
--- msm.orig/arch/arm/mach-msm/perf_trace_counters.h
+++ msm/arch/arm/mach-msm/perf_trace_counters.h
@@ -121,7 +121,7 @@
 
 #endif
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH .
+#define TRACE_INCLUDE_PATH ../mach-msm
 #define TRACE_INCLUDE_FILE perf_trace_counters
 #include <trace/define_trace.h>
 
--- msm.orig/arch/arm/mach-msm/trace_msm_low_power.h
+++ msm/arch/arm/mach-msm/trace_msm_low_power.h
@@ -149,6 +149,6 @@
 );
 #endif
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH .
+#define TRACE_INCLUDE_PATH ../mach-msm
 #define TRACE_INCLUDE_FILE trace_msm_low_power
 #include <trace/define_trace.h>
--- msm.orig/drivers/video/msm/Makefile
+++ msm/drivers/video/msm/Makefile
@@ -173,7 +173,7 @@
 
 obj-$(CONFIG_FB_MSM_TVOUT) += tvout_msm.o
 
-ccflags-y := -I$(src)/mhl
+ccflags-y := -I$(src)/mhl -I$(src)
 obj-$(CONFIG_FB_MSM_HDMI_MHL_8334) += mhl-8334.o
 mhl-8334-objs  += mhl/mhl_8334.o
 mhl-8334-objs  += mhl/mhl_i2c_utils.o
--- msm.orig/arch/arm/mach-msm/trace_rpm_smd.h
+++ msm/arch/arm/mach-msm/trace_rpm_smd.h
@@ -74,6 +74,6 @@
 );
 #endif
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH .
+#define TRACE_INCLUDE_PATH ../mach-msm
 #define TRACE_INCLUDE_FILE trace_rpm_smd
 #include <trace/define_trace.h>
