Signed-off-by: Behan Webster <behanw@converseincode.com>

--- linux.orig/arch/arm64/Makefile
+++ linux/arch/arm64/Makefile
@@ -19,8 +19,16 @@
 
 KBUILD_DEFCONFIG := defconfig
 
-KBUILD_CFLAGS	+= -mgeneral-regs-only
-KBUILD_CPPFLAGS	+= -mlittle-endian
+#ifeq ($(COMPILER),clang)
+# Clang options
+KBUILD_CPPFLAGS	+= $(call cc-option,-target $(CROSS_COMPILE:-=),)
+KBUILD_CFLAGS   += $(call cc-option,-fno-builtin,)
+KBUILD_CFLAGS   += $(call cc-disable-warning,asm-operand-widths)
+KBUILD_CFLAGS   += $(call cc-option,-Xassembler,)
+#endif
+
+KBUILD_CFLAGS	+= $(call cc-option,-mgeneral-regs-only,)
+KBUILD_CPPFLAGS	+= $(call cc-option,-mlittle-endian,)
 AS		+= -EL
 LD		+= -EL
 
