--- android_kernel_samsung_d2.orig/arch/arm/mach-msm/board-m2_att.c
+++ android_kernel_samsung_d2/arch/arm/mach-msm/board-m2_att.c
@@ -1577,7 +1577,7 @@
 	}
 	if (i == 10) {
 		pr_err("%s: fail to get battery ps\n", __func__);
-		return;
+		return ret;
 	}
 #endif
 
@@ -5129,7 +5129,7 @@
 	if (ret) {
 		pr_err("%s: ldo bias gpio %d request"
 				"failed\n", __func__, GPIO_MAIN_MIC_BIAS);
-		return ret;
+		return;
 	}
 	gpio_direction_output(GPIO_MAIN_MIC_BIAS, 0);
 }
--- android_kernel_samsung_d2.orig/arch/arm/mach-msm/board-8960-camera.c
+++ android_kernel_samsung_d2/arch/arm/mach-msm/board-8960-camera.c
@@ -2966,7 +2966,7 @@
 
 		if (rc) {
 			pr_err("%s pmic gpio config failed\n", __func__);
-			return rc;
+			return;
 		}
 		pmic_gpio_msm_flash_cntl_en =
 			PM8921_GPIO_PM_TO_SYS(PMIC_MSM_FLASH_CNTL_EN);
