--- linux-manual.orig/scripts/mod/file2alias.c
+++ linux-manual/scripts/mod/file2alias.c
@@ -116,7 +116,7 @@
                 sprintf(str + strlen(str),                      \
                         sizeof(field) == 1 ? "%02X" :           \
                         sizeof(field) == 2 ? "%04X" :           \
-                        sizeof(field) == 4 ? "%08X" : "",       \
+                        sizeof(field) == 4 ? "%08X" : break,    \
                         field);                                 \
         else                                                    \
                 sprintf(str + strlen(str), "*");                \
