diff --git a/arch/arm/mach-msm/perf_event_msm_krait_l2.c b/arch/arm/mach-msm/perf_event_msm_krait_l2.c
index 65a5d2f..82da286 100644
--- a/arch/arm/mach-msm/perf_event_msm_krait_l2.c
+++ b/arch/arm/mach-msm/perf_event_msm_krait_l2.c
@@ -623,7 +623,7 @@ static int __init register_krait_l2_pmu_driver(void)
 	for (i = 0; i < total_l2_ctrs; i++)
 		disable_counter(i);
 
-	msm_perf_register_l2_reset_callback(&l2_reset_pmu);
+	//msm_perf_register_l2_reset_callback(&l2_reset_pmu);
 	return platform_driver_register(&krait_l2_pmu_driver);
 }
 
