--- msm.orig/drivers/media/platform/msm/camera_v1/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ccflags-y += -Idrivers/media/platform/msm/camera_v1/io
 ccflags-y += -Idrivers/media/platform/msm/camera_v1/vfe
--- msm.orig/drivers/media/platform/msm/camera_v1/actuators/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/actuators/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1/io
 obj-$(CONFIG_MSM_ACTUATOR) += msm_actuator.o
--- msm.orig/drivers/media/platform/msm/camera_v1/cci/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/cci/Makefile
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/platform/msm/camera_v1 -Idrivers/media/platform/msm/camera_v1/server
 obj-$(CONFIG_MSM_CCI) += msm_cci.o
--- msm.orig/drivers/media/platform/msm/camera_v1/csi/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/csi/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/platform/msm/camera_v1 -Idrivers/media/platform/msm/camera_v1/server
 ifeq ($(CONFIG_MSM_CSI20_HEADER),y)
   ccflags-y += -Idrivers/media/platform/msm/camera_v1/csi/include/csi2.0
--- msm.orig/drivers/media/platform/msm/camera_v1/eeprom/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/eeprom/Makefile
@@ -1,5 +1,5 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1/io
 obj-$(CONFIG_MSM_EEPROM) += msm_camera_eeprom.o
 obj-$(CONFIG_IMX074_EEPROM) += imx074_eeprom.o
-obj-$(CONFIG_IMX091_EEPROM) += imx091_eeprom.o
\ No newline at end of file
+obj-$(CONFIG_IMX091_EEPROM) += imx091_eeprom.o
--- msm.orig/drivers/media/platform/msm/camera_v1/flash/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/flash/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/platform/msm/camera_v1
 ccflags-y += -Idrivers/media/platform/msm/camera_v1/io
 obj-$(CONFIG_MSM_CAMERA_FLASH) += msm_flash.o
--- msm.orig/drivers/media/platform/msm/camera_v1/gemini/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/gemini/Makefile
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1
 obj-$(CONFIG_MSM_GEMINI) += msm_gemini_dev.o msm_gemini_sync.o msm_gemini_core.o msm_gemini_hw.o msm_gemini_platform.o
--- msm.orig/drivers/media/platform/msm/camera_v1/io/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/io/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ccflags-y += -Idrivers/media/platform/msm/camera_v1 -Idrivers/media/platform/msm/camera_v1/cci
 obj-$(CONFIG_MSM_CAMERA)   += msm_camera_io_util.o msm_camera_i2c.o
--- msm.orig/drivers/media/platform/msm/camera_v1/mercury/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/mercury/Makefile
@@ -1,3 +1,3 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1
 obj-$(CONFIG_MSM_MERCURY) += msm_mercury_dev.o msm_mercury_core.o msm_mercury_hw.o msm_mercury_platform.o msm_mercury_sync.o
--- msm.orig/drivers/media/platform/msm/camera_v1/sensors/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/sensors/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1/io
 EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1/eeprom
--- msm.orig/drivers/media/platform/msm/camera_v1/server/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/server/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ifeq ($(CONFIG_MSM_CAMERA_V4L2),y)
   EXTRA_CFLAGS += -Idrivers/media/platform/msm/camera_v1
--- msm.orig/drivers/media/platform/msm/camera_v1/vfe/Makefile
+++ msm/drivers/media/platform/msm/camera_v1/vfe/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 ccflags-y += -Idrivers/media/platform/msm/camera_v1
 ccflags-y += -Idrivers/media/platform/msm/camera_v1/server
 ifeq ($(GCC_VERSION),0404)
--- msm.orig/drivers/media/platform/msm/camera_v2/jpeg_10/Makefile
+++ msm/drivers/media/platform/msm/camera_v2/jpeg_10/Makefile
@@ -1,4 +1,4 @@
-GCC_VERSION      := $(shell $(CONFIG_SHELL) $(PWD)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
+GCC_VERSION      := $(shell $(CONFIG_SHELL) $(srctree)/scripts/gcc-version.sh $(CROSS_COMPILE)gcc)
 
 ccflags-y += -Idrivers/media/platform/msm/camera_v2/jpeg_10
 
